### ISSUE TITLE
[Bug] Scraping Filenames with Apostrophe: ' aren't found?

### ISSUE THREAD

**Issue by tundrablasen** [Tikipeter/repository.tikipeter#97]
**tl;dr: Folder Scrapers seems to be fooled by Apostrophe: '**
 
 
First of all I want to salute you @Tikipeter  for this amazing piece of software you have created. If I knew where to donate you some beers (I know you don't want donations) ... I would in a heartbeat.
 
No to the possible bug I might have discovered.
 
**My Setup:**
Vanilla Kodi 19.5
latest Fen Addon
 
Im scraping from two sources. 
1) First Premiumize, 
2) but second I have a home server running with Sonarr and Radarr. So therefore I integrate my local media library via your Folder Scraper into FEN.
 
THANK YOU for Folder Scraper! With this I can integrate my local library with every streaming link via Premiumize, when local media file is not available. Really a Game Changer. into a All in One Solution, and it works perfectly.
 
But It seems, that the Folder Scrapers seems to be fooled by Apostrophe in the Filename. I have of course a limited possibility to test this but I found a pattern with that.
 
My Sonarr is renaming the downloaded files into clean naming Structures. And the Ones with an Apostrophe aren't found via Folder Scraper
 
Examples:
`The.Handmaids.Tale.S05E01.Morning.2160p.HULU.WEB-DL.DDP5.1.HEVC-NTb.mkv`
becomes 
`The Handmaid's Tale (2017) - S05E01 - Morning [HULU WEBDL-2160p][EAC3 5.1][HEVC]-NTb.mkv` 

`Tom.Clancys.Jack.Ryan.S03E07.2160p.AMZN.WEB-DL.DDP5.1.HEVC-NTb.mkv`
becomes
`Tom Clancy's Jack Ryan (2018) - S03E07 - Moscow Rules [AMZN WEBDL-2160p][EAC3 5.1][HEVC]-NTb.mkv`
 
`for.all.mankind.s03e01.2160p.web.h265-glhf.mkv`
becomes
`For All Mankind (2019) - S03E01 - Polaris [WEBDL-2160p][EAC3 Atmos 5.1][h265]-glhf.mkv` 
and so on.
 
As you see, The Handmaid's Tale and Tom Clancy's Jack Ryan aren't found by the Folder Scraper. As I see it, other Filenames in my Local library are fine and detected (without apostrophe?)
 
I reported a year ago another similar bug with the show "And Just Like That…". As I recall it back in the day the "..." in the filename Made similar issues?
 
I know 123Venom helped then with the fix. maybe this time @umbrelladev has a another idea, since you also share coco scrapers (is folder scraper part of cocoscraper?) 

Here is a log:
https://paste.kodi.tv/edequxiduc

As I can see it it does not log the scrape efforts. It just scraped the scrape of Mad men which was locally available (Samba Share from my dietpi)

I hope it helps anyway

**Comment by tundrablasen** [comment]
Update:

i fixed it by myself. I told sonarr and radarr to use a clean naming scheme when renaming the files. therefore all apostrophe are eliminated 

before:
`{Series TitleYear} - S{season:00}E{episode:00} - {Episode CleanTitle} [{Preferred Words }{Quality Full}]{[MediaInfo VideoDynamicRangeType]}{[Mediainfo AudioCodec}{ Mediainfo AudioChannels]}{MediaInfo AudioLanguages}{[MediaInfo VideoCodec]}{-Release Group}`

now:
`{Series CleanTitle} ({Series Year}) - S{season:00}E{episode:00} - {Episode CleanTitle} [{Preferred Words }{Quality Full}]{[MediaInfo VideoDynamicRangeType]}{[Mediainfo AudioCodec}{ Mediainfo AudioChannels]}{MediaInfo AudioLanguages}{[MediaInfo VideoCodec]}{-Release Group}`

results are then:
`The.Handmaids.Tale.S05E01.Morning.2160p.HULU.WEB-DL.DDP5.1.HEVC-NTb.mkv`
becomes
`The Handmaids Tale (2017) - S05E01 - Morning [HULU WEBDL-2160p][EAC3 5.1][HEVC]-NTb.mkv`

and that did the trick.now FENs folder Scraper is finding all local files correctly and shows them in the source select.


to be honest im not so sure anymore if its a bug on fen side, or it just follows a specific naming scheme. 

but I think its still good to point out that apostrophe in the file name kills the source select. if anybody googles, you find here the solution.

**Comment by Tikipeter** [comment]
Glad you sorted it out. Really those results still should have been found, so I'll have a look and see where things are going wrong with the filename cleaning that is done.

**Comment by tundrablasen** [comment]
Thanks for looking it Up.
